sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -107,34 +108,56 @@ public static SpannerAccessor getOrCreate(SpannerConfig spannerConfig) {
   private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     SpannerOptions.Builder builder = SpannerOptions.newBuilder();
 
-    ValueProvider<Duration> commitDeadline = spannerConfig.getCommitDeadline();
-    if (commitDeadline != null && commitDeadline.get().getMillis() > 0) {
+    // Set retryable codes
+    if (spannerConfig.getRetryableCodes() != null) {
```

Comment:
I think that it might be unintuitive that the retryable codes set by the user apply only to the commit settings and the executeStreamingSqlSettings.

Is there any way we could explore settings for specific methods in the `SpannerConfig`? Maybe a mapping of `Method` to retryable codes / retry settings?

Reply:
I am not sure how to model this yet, but if we do it by creating a method for each setting, it might become too large, so I was thinking that maybe we could pass in a `SpannerConfig.setSpannerStubSettings()` and extract all the settings from that.

Reply:
I actually started this work from trying to put in `SpannerStubSettings` rather than the separate `retryableCodes` and `retrySettings` currently here, but `SpannerStubSettings` is not serialisable and I couldn't get it to work after trying for hours.

I have changed this part so that the retryable codes get applied to all API methods rather than just the commit and executeStreamingSql though. Let me know if that doesn't work :)

Reply:
Makes sense, thanks for trying. I think this approach is nice.